### PR TITLE
Document the built-in patterns in the Technical Reference section

### PR DESCRIPTION
**Pull request by seisman**
**Preview**:

- https://pygmt-dev--3466.org.readthedocs.build/en/3466/techref/patterns.html
- https://pygmt-dev--3466.org.readthedocs.build/en/3466/gallery/symbols/patterns.html

### CODE REVIEW

File: examples/gallery/symbols/patterns.py [hunk 1]
```diff
@@ -1,67 +1,44 @@
...
-**b**\ ackground and **f**\ oreground colors (by appending **+b**\ *color*
-and/or **+f**\ *color*) that will replace the default white and black pixels,
-respectively. Excluding *color* from a fore- or background specification yields
-a transparent image where only the back- or foreground pixels will be painted.
-The **+r**\ *dpi* modifier sets the resolution in dpi.
+GMT provides 90 predefined patterns that can be used in PyGMT. The patterns are numbered
+from 1 to 90. The patterns can be colored and inverted. The resolution of the pattern
+can be changed, and the background and foreground colors can be set. For a complete list
+of available patterns and the full syntax to specify a pattern, refer to the
+{doc}`/techref/patterns`.
```
Comment (seisman) [inline examples/gallery/symbols/patterns.py:26]:
```suggestion
:doc:`/techref/patterns`.
```

File: examples/gallery/symbols/patterns.py [hunk 2]
```diff
@@ -1,67 +1,44 @@
-r"""
+"""
 Bit and hachure patterns
 ========================
 
-PyGMT allows using bit or hachure patterns via the ``fill`` parameter
-or similar parameters:
+In additon to solid colors, PyGMT also allows using bit or hachure patterns to fill
```
Comment (yvonnefroehlich) [inline examples/gallery/symbols/patterns.py:5]:
```suggestion
In addition to solid colors, PyGMT also allows using bit or hachure patterns to fill
```

**Review by yvonnefroehlich — COMMENTED** [review]
Nice addition to the "Technical Reference" section. Will look at this PR tomorrow (or in the next days) in more detail.

File: examples/gallery/symbols/patterns.py [hunk 3]
```diff
@@ -1,67 +1,44 @@
...
+"""
 Bit and hachure patterns
 ========================
 
-PyGMT allows using bit or hachure patterns via the ``fill`` parameter
-or similar parameters:
+In addition to solid colors, PyGMT also allows using bit or hachure patterns to fill
+symbols, polygons, and other areas, via the ``fill`` parameter or similar parameters.
+
+Example function parameters that support bit and hachure patterns include:
```
Comment (yvonnefroehlich) [inline examples/gallery/symbols/patterns.py:8]:
```suggestion
Example method parameters that support bit and hachure patterns include:
```

File: examples/gallery/symbols/patterns.py [hunk 4]
```diff
@@ -1,67 +1,44 @@
-r"""
+"""
 Bit and hachure patterns
 ========================
 
-PyGMT allows using bit or hachure patterns via the ``fill`` parameter
-or similar parameters:
+In addition to solid colors, PyGMT also allows using bit or hachure patterns to fill
```
Comment (yvonnefroehlich) [inline examples/gallery/symbols/patterns.py:5]:
```suggestion
In addition to solid colors, PyGMT also allows using bit and hachure patterns to fill
```

File: examples/gallery/symbols/patterns.py [hunk 5]
```diff
@@ -1,67 +1,44 @@
...
-**b**\ ackground and **f**\ oreground colors (by appending **+b**\ *color*
-and/or **+f**\ *color*) that will replace the default white and black pixels,
-respectively. Excluding *color* from a fore- or background specification yields
-a transparent image where only the back- or foreground pixels will be painted.
-The **+r**\ *dpi* modifier sets the resolution in dpi.
+GMT provides 90 predefined patterns that can be used in PyGMT. The patterns are numbered
+from 1 to 90. The patterns can be colored and inverted. The resolution of the pattern
+can be changed, and the background and foreground colors can be set. For a complete list
+of available patterns and the full syntax to specify a pattern, refer to the
+:doc:`/techref/patterns`.
```
Comment (yvonnefroehlich) [inline examples/gallery/symbols/patterns.py:26]:
```suggestion
GMT provides 90 predefined patterns that can be used in PyGMT. The patterns are numbered
from 1 to 90, and can be colored and inverted. The resolution of the pattern
can be changed, and the background and foreground colors can be set. For a complete list
of available patterns and the full syntax to specify a pattern, refer to the
:doc:`/techref/patterns`.
```

File: examples/gallery/symbols/patterns.py [hunk 6]
```diff
@@ -1,67 +1,44 @@
-r"""
+"""
 Bit and hachure patterns
 ========================
 
-PyGMT allows using bit or hachure patterns via the ``fill`` parameter
-or similar parameters:
+In addition to solid colors, PyGMT also allows using bit and hachure patterns to fill
```
Comment (yvonnefroehlich) [inline examples/gallery/symbols/patterns.py:5]: I am personally prefer removing "solid", as it sounds a little bit strange to me (there are no fluid colors...).

```suggestion
In addition to colors, PyGMT also allows using bit and hachure patterns to fill
```

**Review by yvonnefroehlich — APPROVED** [review]
Looks good to me (besides the minor comment)!

File: examples/gallery/symbols/patterns.py [hunk 7]
```diff
@@ -70,23 +47,21 @@
...
+fig.basemap(
+    region=[0, 10, 0, 12],
+    projection="X10c",
+    frame="rlbt+glightgray+tBit and Hachure Patterns",
+)
+
+y = 11
+for pattern in patterns:
+    # Plot a square with the pattern as fill.
+    # The square has a size of 2 centimeters with 1 point thick, black outline.
```
Comment (michaelgrund) [inline examples/gallery/symbols/patterns.py:62]:
```suggestion
    # The square has a size of 2 centimeters with a 1 point thick, black outline.
```

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**